docs/sources/v0.51.x/using-k6-browser/migrating-to-k6-v0-52.md

```diff
@@ -0,0 +1,215 @@
+---
+title: 'Migrating to k6 v0.52'
+description: 'A migration guide to ease the process of transitioning to the new k6 browser module version'
+weight: 04
+---
+
+# Migrating to k6 v0.52
+
+This guide outlines the key changes you will need to make when moving your existing k6 browser test scripts to the **upcoming** latest [k6 browser module](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser) (bundled with [k6 version 0.52](https://github.com/grafana/k6/releases/tag/v0.52.0)).
+
+## Key changes
+
+In the upcoming release of k6, we will be introducing a breaking change that will affect **all scripts** that use the k6 browser module. The main breaking changes are:
+
+1. Converting majority of the k6 browser module APIs to asynchronous (async) APIs. What this means is that they will return a `promise` that will `resolve` to a value when the API call succeeds, or it will `reject` when an error occurs.
+2. A side affect of making this async change, is that you will need to use a workaround to work with the k6 [check]({{< relref "../javascript-api/k6/check" >}}) API.
+3. Finally, it's also worth reiterating that the [group]({{< relref "../javascript-api/k6/group" >}}) API still doesn't work with async APIs.
+
+If you are interested in the rational as to why we're making this change, have a look at the [v0.51 release notes](https://github.com/grafana/k6/releases/tag/v0.51.0).
+
+## Migrating to async
+
+To ensure your scripts work with the upcoming release of k6, you will need to ensure that all the synchronous (sync) APIs that are going to be migrated over to async are handled appropriately. In majority of cases you will only need to add `await` in front of the API call, like so:
+
+From
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = browser.newPage();
+  // ...
+}
+```
+
+{{< /code >}}
+
+To
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = await browser.newPage();
+  // ...
+}
+```
+
+{{< /code >}}
+
+You will likely already have encountered async APIs when working with the browser module such as [page.click]({{< relref "../javascript-api/k6-experimental/browser/page/click/" >}}), so the use of `async` and `await` keywords will be familiar to you.
+
+The full list of APIs that will migrate over to be async can be found in this [github comment](https://github.com/grafana/xk6-browser/issues/428#issuecomment-1964020837). You will also find a screenshot of a comparison between the [fillform.js example](https://github.com/grafana/xk6-browser/blob/main/examples/fillform.js) in `v0.51` and `v0.52` in another [comment](https://github.com/grafana/xk6-browser/issues/428#issuecomment-2141634960) in the same github issue.
```

Comment:
I would put that list and image here in this migration document for immediate visibility instead of linking it to a comment on a Github issue.

Reply:
Good spot, i was supposed to put a comment here asking for suggestions on this.

@heitortsergent what's the best way to add this very long list to the docs? Is it possible to have a collapsible list?

I'm assuming the image is straight forward. However it would be better if there is already a built in code comparison view?

Reply:
I've added the full list of APIs in 7b5f2d7c71003ce817231212de3a5cea997011c0

I've added a new screenshot comparing a simpler example script in 410edd5e7562f9099779073928be73dab12b7478

Reply:
I've found a way to collapse long lists.